packages/cli/src/executions/legacy-sqlite-execution-recovery.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Logger } from '@n8n/backend-common';
+import { GlobalConfig } from '@n8n/config';
+import { ExecutionData, ExecutionRepository } from '@n8n/db';
+import { Service } from '@n8n/di';
+
+/**
+ * Service for recovering executions that are missing execution data, this should only happen
+ * for sqlite legacy databases.
+ */
+@Service()
+export class LegacySqliteExecutionRecoveryService {
+	private readonly legacySqlite: boolean;
+
+	constructor(
+		private readonly logger: Logger,
+		private readonly executionRepository: ExecutionRepository,
+		private readonly globalConfig: GlobalConfig,
+	) {
+		const { type: dbType, sqlite: sqliteConfig } = this.globalConfig.database;
+		this.legacySqlite = dbType === 'sqlite' && sqliteConfig.poolSize === 0;
+	}
+
+	/**
+	 * Recover key properties of a truncated execution using event logs.
+	 */
+	async cleanupWorkflowExecutions() {
+		if (!this.legacySqlite) {
+			return;
+		}
+
+		this.logger.info('Starting legacy SQLite execution recovery...');
+
+		const invalidExecutions = await this.executionRepository
+			.createQueryBuilder('execution')
+			.where('execution.status = :status', { status: 'new' })
+			.andWhere(
+				'NOT EXISTS (' +
+					this.executionRepository.manager
+						.createQueryBuilder()
+						.select('1')
+						.from(ExecutionData, 'execution_data')
+						.where('execution_data.executionId = execution.id')
+						.getQuery() +
+					')',
+			)
+			.select('execution.id')
+			.getMany();
+
+		if (invalidExecutions.length > 0) {
+			await this.executionRepository.markAsCrashed(invalidExecutions.map((e) => e.id));
```

Comment:
Should we mark them as `crashed` or delete them? 
- If the user tries to inspect a dataless `crashed` execution on the UI, does the UI work? Or if the user tries to manually delete a dataless `crashed` execution, the deletion works?
- If we mark them as `crashed` here, does the execution recovery service try to recover them? We should recover only once.
- If the query returns duplicate execution IDs, we should deduplicate before the update for less DB work.